server/server.go

```diff
@@ -100,9 +127,28 @@ func New(cfg Config) (*Server, error) {
 	cfg.Mux.HandleFunc("/api/delete", ret.deleteSecret)
 	cfg.Mux.HandleFunc("/api/delete-version", ret.deleteVersion)
 
+	if ret.backupClient != nil {
+		go ret.periodicBackup()
+	}
+
 	return ret, nil
 }
 
+func makeS3Client(assumeRole string) (*s3.Client, error) {
```

Comment:
Plumbed env.Context() throughout. The shutdown isn't terribly orderly right now in that the backup goroutine doesn't synchronize with general shutdown in main, but it's good enough that if we spawn an in-memory server for tests or something and cancel its context, it will promptly quiesce.

Reply:
Well for tests we probably won't start this goroutine anyway I suppose.